src/tests/ftest/osa/osa_offline_extend.py

```diff
@@ -0,0 +1,196 @@
+        self.extra_servers = self.params.get("test_servers",
+                                             "/run/extra_servers/*")
+
+    @fail_on(CommandFailure)
+    def get_pool_leader(self):
+        """Get the pool leader.
+
+        Returns:
+            int: pool leader value
+
+        """
+        data = self.dmg_command.pool_query(self.pool.uuid)
+        return int(data["leader"])
+
+    @fail_on(CommandFailure)
+    def get_pool_version(self):
+        """Get the pool version.
+
+        Returns:
+            int: pool_version_value
+
+        """
+        data = self.dmg_command.pool_query(self.pool.uuid)
+        return int(data["version"])
+
+    @fail_on(DaosApiError)
+    def write_single_object(self):
+        """Write some data to the existing pool."""
+        self.pool.connect(2)
+        csum = self.params.get("enable_checksum", '/run/container/*')
+        container = DaosContainer(self.context)
+        input_param = container.cont_input_values
+        input_param.enable_chksum = csum
+        container.create(poh=self.pool.pool.handle,
+                         con_prop=input_param)
+        container.open()
+        obj = DaosObj(self.context, container)
+        obj.create(objcls=1)
+        obj.open()
+        ioreq = IORequest(self.context,
+                          container,
+                          obj, objtype=4)
+        self.log.info("Writing the Single Dataset")
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length)
+                d_key_value = "dkey {0}".format(dkey)
+                c_dkey = ctypes.create_string_buffer(d_key_value)
+                a_key_value = "akey {0}".format(akey)
+                c_akey = ctypes.create_string_buffer(a_key_value)
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+                ioreq.single_insert(c_dkey, c_akey, c_value, c_size)
+
+    def run_offline_extend_test(self, num_pool, data=False):
+        """Run the offline extend without data.
+            Args:
+            num_pool (int) : total pools to create for testing purposes.
+            data (bool) : whether pool has no data or to create
+                          some data in pool. Defaults to False.
+        """
+        # Create a pool
+        pool = {}
+        pool_uuid = []
+        target_list = []
+        total_servers = len(self.hostlist_servers)
+
+        # Exclude target : random two targets
+        n = random.randint(1, 7)
+        target_list.append(n)
+        target_list.append(n+1)
+
+        # Extend a rank (or server)
+        # rank index starts from zero
+        rank = total_servers
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
```

Comment:
```suggestion
                                 dmg_command=self.dmg_command)
```
Can't we use the value set in setUp?

Reply:
Agree. Will do the change.

Reply:
Done.